### PR TITLE
feat(tree,tree-item): update indentation

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #7096 

## Summary
Updates padding indentation for `tree-item`'s in `tree`

### CODE REVIEW

File: packages/calcite-components/src/components/tree-item/tree-item.tsx [hunk 1]
```diff
@@ -462,8 +462,13 @@ export class TreeItem extends LitElement implements InteractiveComponent {
             >
               {chevron}
               {itemIndicator}
+              {checkbox ? checkbox : null}
               {this.iconStart ? iconStartEl : null}
-              {checkbox ? checkbox : defaultSlotNode}
+              {checkbox ? (
+                <label class={CSS.checkboxLabel}>{defaultSlotNode}</label>
+              ) : (
+                defaultSlotNode
```
Comment (anveshmekala) [inline packages/calcite-components/src/components/tree-item/tree-item.tsx:470]: No impact for AT users with VoiceOver. Adding @geospatialem  for review.

**Review by ashetland — APPROVED** [review]
LGTM! 🚀

**Comment by anveshmekala** [comment]
This PR  made changes to stories to include all `scale`s and `selectionMode`s . @macandcheese  Please review if any common use case  is missing or would be beneficial to add a screenshot test. Thanks!

File: packages/calcite-components/src/components/tree-item/tree-item.scss [hunk 2]
```diff
@@ -1,80 +1,53 @@
+:host([scale="s"]) {
+  --calcite-internal-tree-item-spacing-unit: 0.25rem;
```
[the remark this answers is not in the document]
Comment (anveshmekala) [reply packages/calcite-components/src/components/tree-item/tree-item.scss:2]: AFAIK, it is recommended to leave them as `rem` or `px` for now and update them to tokens in a separate effort once tokenization is complete. Please correct me if wrong @alisonailea

**Pull request merged**